### ISSUE TITLE
:help does nothing

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura#306]
*On GitLab by @shakham on Jan 21, 2022, 18:27*

---

System: Clear Linux 35670

zathura 0.4.8

girara 0.3.6 (runtime: 0.3.6)

(plugin) pdf-poppler (0.3.0) (/usr/lib64/zathura/libpdf-poppler.so)

**Comment by sebastinas** [comment]
*On GitLab by @kmdodrill on Jan 25, 2022, 20:13*

---

I noticed this as well and it seems as though the help command hasn't done anything since it was initialized in 2011. https://git.pwmt.org/pwmt/zathura/-/commit/bb466fc8912678936016609e7c182b448b35f11e

I guess the question is: what should the help command output? I'm wondering if something was holding it back.

**Comment by sebastinas** [comment]
*On GitLab by @shakham on Jan 26, 2022, 03:02*

---

what about the man pages ?

**Comment by sebastinas** [comment]
*On GitLab by @kmdodrill on Jan 28, 2022, 04:15*

---

Yeah, I'd say maybe listing out the key bindings and what they do would be good. It's likely that it's best to consider custom keybinds. I took a look through the code, but I'm not sure how you would get the values of custom keybinds - perhaps through the values set in config.c or something. I'll look into it some more and maybe try to figure it out.

**Comment by sebastinas** [comment]
*On GitLab by @kmdodrill on Feb 1, 2022, 24:20*

---

I have looked into this some and found that `session->bindings.shortcuts` should suffice for grabbing the keybindings, but I'm not sure how to loop through them. I'm trying to do something like:

```c
GIRARA_LIST_FOREACH_BODY(session->bindings.shortcuts, girara_shortcut_t*, shortcut,
  g_string_append_printf(string, "<b>%d</b>: %d\n", shortcut->mask, shortcut->key);
);
```

but I get `error: invalid use of typedef 'girara_shortcut_t'` for both `shortcut->mask` and `shortcut->key`. I'm new to C - guessing I'm missing an include or something like that but not sure. Any help is appreciated.

EDIT: As far as I can tell, it's just that `girara_shortcut_t` is internal to girara, therefore not available as a complete type in zathura. Again, I'm still new to C so I'm not 100% sure on this. Basically, I don't see a way to get the keybinds unless girara is changed to where `girara_shortcut_t` is no longer internal.

**Comment by sebastinas** [comment]
Fixed in 61d436a.
